src/components/Context/createComponent.js

```diff
@@ -1,39 +1,54 @@
-export default (config, logger, availableContexts, defaultContexts) => {
-  let configuredContexts = {};
-  const onBeforeRequest = payload => {
-    const context = Object.keys(configuredContexts).reduce((memo, key) => {
-      memo = { ...memo, ...configuredContexts[key]() }; // eslint-disable-line no-param-reassign
-      return memo;
-    }, {});
-    payload.addContext(context);
-  };
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+export default (config, logger, availableContexts, defaultContextNames) => {
```

Comment:
Changing `defaultContexts` (an array of the context providers) to `defaultContextNames` (an array of strings) helped align the value with what the user would provide through configuration (an array of strings), which makes the logic for falling back to the default contexts more straightforward IMO.

Reply:
👍